mlflow/genai/judges/adapters/litellm_adapter.py

```diff
@@ -197,26 +207,58 @@ def _invoke_litellm_and_handle_tools(
 
     messages = [litellm.Message(role=msg.role, content=msg.content) for msg in messages]
 
-    litellm_model_uri = f"{provider}/{model_name}"
-    tools = []
+    # Construct model URI and gateway params
+    if provider == "gateway":
+        from mlflow.tracking import get_tracking_uri
+        from mlflow.utils.uri import append_to_uri_path, is_http_uri
+
+        tracking_uri = get_tracking_uri()
+
+        # Validate that tracking URI is a valid HTTP(S) URL for gateway
+        if not is_http_uri(tracking_uri):
+            raise MlflowException(
+                f"Gateway provider requires an HTTP(S) tracking URI, but got: '{tracking_uri}'. "
+                "The gateway provider routes requests through the MLflow tracking server. "
+                "Please set MLFLOW_TRACKING_URI to a valid HTTP(S) URL "
+                "(e.g., 'http://localhost:5000' or 'https://your-mlflow-server.com')."
+            )
 
+        api_base = append_to_uri_path(tracking_uri, "gateway/mlflow/v1/")
+
+        # Use openai/ prefix for LiteLLM to use OpenAI-compatible format.
+        # LiteLLM strips the prefix, so gateway receives model_name as the endpoint.
+        model = f"openai/{model_name}"
+        # LiteLLM requires api_key to be set when using custom api_base, otherwise it
+        # raises AuthenticationError looking for OPENAI_API_KEY env var. Gateway handles
+        # auth in the server layer, so we pass a dummy value to satisfy LiteLLM.
+        api_key = "mlflow-gateway-auth"
+    else:
+        model = f"{provider}/{model_name}"
+        api_base = None
+        api_key = None
+
+    tools = []
     if trace is not None:
         judge_tools = list_judge_tools()
         tools = [tool.get_definition().to_dict() for tool in judge_tools]
 
-    def _prune_messages_for_context_window():
+    def _prune_messages_for_context_window() -> list[litellm.Message] | None:
+        if provider == "gateway":
+            # For gateway provider, we don't know the underlying model,
+            # so simply remove the oldest tool call pair.
+            return _prune_messages_exceeding_context_window_length(messages)
+
+        # For direct providers, use token-counting based pruning.
         try:
-            max_context_length = litellm.get_max_tokens(litellm_model_uri)
+            max_context_length = litellm.get_max_tokens(model)
         except Exception:
             max_context_length = None
 
         return _prune_messages_exceeding_context_window_length(
-            messages=messages,
-            model=litellm_model_uri,
-            max_tokens=max_context_length or 100000,
+            messages, model=model, max_tokens=max_context_length or 100000
```

Comment:
note: Unrelated to this PR, but we should probably stop using a const value (100000) as the fallback. Current logic hits the same error for models whose context window is less than 100000 until we hit max iteration